app/services/otp_rate_limiter.rb

```diff
@@ -15,19 +15,14 @@ def exceeded_otp_send_limit?
   end
 
   def max_requests_reached?
-    return throttle.throttled? if IdentityConfig.store.redis_throttle_otp_rate_limiter_read_enabled
-
-    entry_for_current_phone.otp_send_count > otp_maxretry_times
+    return throttle.throttled?
```

Comment:
Can drop the `return` ?

```suggestion
    throttle.throttled?
```